openoa/utils/downloader.py

```diff
@@ -330,6 +340,200 @@ def get_era5_monthly(
+
+    # assign end_year to current year if not provided by the user
+    if end_date is None:
+        end_date = f"{now.year}-{now.month:02}-{now.day:02}"
+
+    # convert dates to datetime objects
+    start_date = datetime.datetime.strptime(start_date, "%Y-%m-%d")
+    end_date = datetime.datetime.strptime(end_date, "%Y-%m-%d")
+    end_date += datetime.timedelta(hours=23, minutes=59)  # include all times in last day
+
+    # check that the start and end dates are the right way around
+    if start_date > end_date:
+        logger.error("The start_date should be less than or equal to the end_date")
+        logger.error(f"start_date = {start_date.date()}, end_date = {end_date.date()}")
+        raise ValueError("The start_date should be less than or equal to the end_date")
+
+    # list all years that will be downloaded
+    years = list(range(start_date.year, end_date.year + 1))
+
+    # find the nearest coordinate grid point
+    node_spacing = 0.25
+    lat_nearest = node_spacing * np.round(lat / node_spacing)
+    lon_nearest = node_spacing * np.round(lon / node_spacing)
+
+    # Get data for single grid point
+    # See: https://cds.climate.copernicus.eu/datasets/reanalysis-era5-single-levels?tab=form
+    # for formulating other requests from cds
+    cds_dataset = "reanalysis-era5-single-levels"
+    cds_request = {
+        "product_type": "reanalysis",
+        "format": "netcdf",
+        "variable": [
+            "100m_u_component_of_wind",
+            "100m_v_component_of_wind",
+            "2m_temperature",
+            "surface_pressure",
+        ],
+        "year": None,
+        "month": None,
+        "day": [f"{i:02d}" for i in range(1, 32)],
+        "time": [f"{i:02d}:00" for i in range(24)],
+        "product_type": "reanalysis",
+        "area": [
+            lat_nearest + 0.01,
+            lon_nearest,
+            lat_nearest,
+            lon_nearest + 0.01,
+        ],
+    }
+
+    # download the data
+    for year in years:
+        outfile = save_pathname / f"{save_filename}_{year}.nc"
+
+        # limit to months of interest
+        if year == start_date.year:
+            if year == end_date.year:
+                months = list(range(start_date.month, end_date.month + 1))
+            else:
+                months = list(range(start_date.month, 12 + 1))
+        elif year == end_date.year:
+            months = list(range(1, end_date.month + 1))
+        else:
+            months = list(range(1, 12 + 1))
+
+        if not outfile.is_file():
+            logger.info(f"Downloading ERA5: {outfile}")
+
+            try:
+                cds_request.update({"year": year, "month": months})
+                c.retrieve(cds_dataset, cds_request, outfile)
+            except Exception as e:
+                logger.error(f"Failed to download ERA5: {outfile}")
+                logger.error(e)
+
+    # get the saved data
+    ds_nc = xr.open_mfdataset(f"{save_pathname / f'{save_filename}*.nc'}")
```

Comment:
This works, but just a note that xarray, pandas, and pretty much any major package will handle `Path` object file inputs at this point.

Reply:
It looks like that array function will only accept glob-like file paths if they are strings. I get an error when leaving it as a `Path` object. I cleaned it up a little by just wrapping `str()` around the `Path` object.